### PR TITLE
13748: Additional logging

### PR DESCRIPTION
**Pull request by benkitzelman**
We want to be explicit when insight transactions are skipped. This PR adds logging when a transaction is skipped due to disabled config flag

### CODE REVIEW

File: lib/insight_client/base.rb [hunk 1]
```diff
@@ -42,6 +41,18 @@ def _get(path, data)
       def full_uri(path)
         config.base_uri + path
       end
+
+      protected
```
Comment (johncarney) [inline lib/insight_client/base.rb]: Why `protected` instead of `private`?

File: lib/insight_client/base.rb [hunk 2]
```diff
@@ -12,19 +12,18 @@ def config
...
+        do_transaction(request_method, url, data)
       end
 
       def transaction!(request_method, url, data = {})
-        return {} unless config.enabled
-
-        response = send(request_method, url, data)
-        raise InsightClient::Error.new("#{response.code}: #{response.response.body}") unless response.success?
-        JSON.parse(response.body)
+        do_transaction(request_method, url, data) do | response |
```
Comment (johncarney) [inline lib/insight_client/base.rb]: Remove the whitespace around `response` in the block parameters.

File: lib/insight_client/base.rb [hunk 3]
```diff
@@ -12,19 +12,18 @@ def config
...
+        config.logger
+      end
 
-        response = send(request_method, url, data)
-        JSON.parse(response.body)
+      def transaction(request_method, url, data = {})
+        do_transaction(request_method, url, data)
       end
 
       def transaction!(request_method, url, data = {})
```
Comment (johncarney) [inline lib/insight_client/base.rb]: Are there any tests that assert that `transaction!` will raise an error, but `transaction` won't?

File: lib/insight_client/base.rb (hunk 1 above)
Comment (benkitzelman) [inline lib/insight_client/base.rb]: Oops - I forgot about this PR... I just figured it made sense to allow extending classes to access / overwrite this if needed. I guess when / if its needed ppl can change then. I'll set it to private

File: lib/insight_client/base.rb (hunk 3 above)
Comment (benkitzelman) [inline lib/insight_client/base.rb]: there are now

File: lib/insight_client/base.rb (hunk 1 above)
Comment (johncarney) [inline lib/insight_client/base.rb]: `protected` doesn't work that way. Any subclass can access or overwrite `private` methods. `private` methods can only be accessed by `self`. `protected` methods can be called by any instance of the defining class or its subclasses. This is mostly only useful for implementing operators.
```ruby
class MyClass
  # stuff ...

  def <=>(other)
    comparable_value <=> other.comparable_value
  end

  protected
    def comparable_value
      # return some value
    end
end
```
In the above example, if `comparable_value` was private, the `<=>` operator would not work.

**Pull request merged**